src/components/App/App.js

```diff
@@ -0,0 +1,312 @@
+        // console.log("Sign Up currentUser is:", currentUser);
+        console.log(userData.name, "you are now a special member");
+        setSignupPopupOpen(false);
+      }
+    } catch (err) {
+      console.log(err, err.response, err.status);
+      if (err) {
+        setConflictErr(true);
+        setTimeout(function () {
+          setConflictErr(false);
+        }, 2000);
+      }
+      setIsTooltipOpen(false);
+      return;
+    }
+  }
+
+  // login:
+  async function handleSigninSubmit(data) {
+    try {
+      //mainApi sets up the "jwt" token:
+      const token = await mainApi.signin(data);
+      if (token) {
+        setLoggedIn(true);
+        window.location.reload();
+        setSigninPopupOpen(false);
+      }
+    } catch (res) {
+      window.alert("You have entered an invalid username or password");
+      return;
+    }
+  }
+
+  function handleLogOut() {
+    navigate("/");
+    localStorage.removeItem("jwt");
+    localStorage.removeItem("keyword");
+    localStorage.removeItem("articles");
+
+    setLoggedIn(false);
+    setShowArticleSection(false);
+  }
+
+  // ========>>>  CARDS FUNCTIONALITY ==========//
+
+  // search for news by keyword
+  async function handleSearchQuery(event, topic) {
+    try {
+      event.preventDefault();
+      setPreloader(true);
+      console.log(topic);
+      let searchRes = await getNewsApi(topic);
+      if (searchRes.articles.length === 0) {
+        setNotFound(true);
+        localStorage.removeItem("keyword");
+        setShowArticleSection(false);
+      }
+      if (searchRes.articles.length !== 0) {
+        //close if prev not found is true:
+        setNotFound(false);
+        setPreloader(false);
+        setSearchedArticles(searchRes.articles);
+        setShowArticleSection(true);
+        //set local value for card count:
+        localStorage.setItem("card-count", 3);
+        console.log(searchRes);
+        //store cards locally:
+        localStorage.setItem("articles", JSON.stringify(searchRes.articles));
+      }
+    } catch (err) {
+      localStorage.removeItem("articles");
+      alert("The search has failed.");
+      console.log("Search query error:", err);
+    } finally {
+      setPreloader(false);
+    }
+  }
+
+  // save article by user
+  async function handleSaveArticle(article) {
+    try {
+      mainApi.saveArticle(article);
+      const userArticles = await mainApi.getSavedArticles();
+      if (userArticles) {
+        setSavedArticles(userArticles);
+      }
+    } catch (err) {
+      console.log(err);
+      if (!loggedIn) console.log("error while saving article");
+    }
+  }
+
+  async function handleDeleteArticle(article) {
+    try {
+      await mainApi.deleteSavedArticle(article);
+      // console.log(article);
+      const userArticles = await mainApi.getSavedArticles();
+      if (userArticles) {
+        setSavedArticles(userArticles);
+      }
+    } catch (err) {
+      console.log(err);
+      window.location.reload();
```

Comment:
`window.location.reload();` please, delete it from your project because React should work without any reloadings. It's the main purpose of SPA-apps